grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy

```diff
@@ -1231,7 +1231,10 @@ class InventoryService implements ApplicationContextAware {
                     def quantityOnHand = quantityBinLocationMap[product][inventoryItem][binLocation]
 
                     def quantityAvailableMap = quantityAvailableInventoryItemMap[product][inventoryItem]
-                    def quantityAvailable = quantityAvailableMap ? quantityAvailableMap[binLocation] : 0
+                    def quantityAvailable =
+                        quantityAvailableMap
+                            ? quantityAvailableMap.find { key, value -> key?.id == binLocation?.id }?.value
```

Comment:
Given that this is probably the most important (and worst performing) algorithm in the system (my bad), I'm not sure we want to change the lookup from constant (1) to linear (N) here. 

Is there another way to deal with this?

Reply:
I think taking `binLocation?.id` (or `null`) as a key might do the work here. Depending on how much you have to refactor.

Reply:
Fortunately after diving into the "source" of that Map it turned out that I could've safely changed the key to be the `id`, not the whole object.